src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -248,6 +325,35 @@ public void testUserLifecycle() throws Exception {
     }
   }
 
+  @Test
+  public void testLastRefreshTime() throws Exception {
+    RandomUser user = RandomUser.create();
+    UserRecord newUserRecord = auth.createUser(new CreateRequest()
+        .setUid(user.uid)
+        .setEmail(user.email)
+        .setEmailVerified(false)
+        .setPassword("password"));
+
+    try {
+      // New users should not have a lastRefreshTimestamp set.
+      assertEquals(0, newUserRecord.getUserMetadata().getLastRefreshTimestamp());
+
+      // Login to cause the lastRefreshTimestamp to be set.
+      signInWithPassword(newUserRecord.getEmail(), "password");
+
+      UserRecord userRecord = auth.getUser(newUserRecord.getUid());
+
+      // Ensure the lastRefreshTimestamp is approximately "now" (with a tollerance of 30 minutes).
```

Comment:
30 minutes sound too high. Did you mean 30 seconds?

Reply:
I did mean 30m. I considered something shorter (such as 30s) but I wanted to allow a bit of clock skew as well as time for an extra long context switch on an overloaded CI server. (Both of which have bitten me in the past.) 30m seems like it should be more than long enough to cover both goals, but short enough that 
a) a random value is highly unlikely to fall within the range.
b) timezone incorrectness should still cause a failure. Hm. Well, possibly except for "interesting" timezones (i.e. non-hour increments from UTC), I suppose. I've reduced the timeout from 30m to 10m, which should put us beneath a timezone error from a +Xh:45m timezone.